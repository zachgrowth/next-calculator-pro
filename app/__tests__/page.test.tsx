import '@testing-library/jest-dom'
import { render, screen, within } from '@testing-library/react'
import Home from '../page'

describe('Home Page', () => {
  beforeEach(() => {
    render(<Home />)
  })

  describe('Hero Section', () => {
    it('renders main heading correctly', () => {
      const heroSection = screen.getByTestId('hero-section')
      const mainHeading = within(heroSection).getByRole('heading', { 
        name: /专业的健康计算工具/i,
        level: 1 
      })
      expect(mainHeading).toBeInTheDocument()
    })
  })

  describe('Features Section', () => {
    it('renders feature sections correctly', () => {
      const featuresSection = screen.getByTestId('features-section')
      expect(featuresSection).toBeInTheDocument()
      expect(featuresSection).toHaveAttribute('aria-label', '特性介绍')
      
      const features = [
        { title: '精确计算', testId: 'feature-card-precise' },
        { title: '健康建议', testId: 'feature-card-advice' },
        { title: '科学依据', testId: 'feature-card-science' }
      ]

      features.forEach(({ title, testId }) => {
        const featureCard = screen.getByTestId(testId)
        expect(featureCard).toBeInTheDocument()
        expect(featureCard).toHaveAttribute('aria-label', `${title}特性`)
        
        const heading = within(featureCard).getByRole('heading', { 
          name: title,
          level: 3 
        })
        expect(heading).toBeInTheDocument()
      })
    })
  })

  describe('Calculators Section', () => {
    it('renders calculator sections correctly', () => {
      const calculatorsSection = screen.getByTestId('calculators-section')
      expect(calculatorsSection).toBeInTheDocument()
      expect(calculatorsSection).toHaveAttribute('aria-label', 'calculators')
      
      const sectionHeading = within(calculatorsSection).getByRole('heading', {
        name: /选择计算工具/i,
        level: 2
      })
      expect(sectionHeading).toBeInTheDocument()

      const calculators = [
        { title: '体脂计算器', testId: 'calculator-card-体脂计算器' },
        { title: 'BMI计算器', testId: 'calculator-card-BMI计算器' },
        { title: '卡路里计算器', testId: 'calculator-card-卡路里计算器' }
      ]

      calculators.forEach(({ title, testId }) => {
        const calculatorCard = screen.getByTestId(testId)
        expect(calculatorCard).toBeInTheDocument()
        expect(calculatorCard).toHaveAttribute('aria-label', `${title}卡片`)
        
        const heading = within(calculatorCard).getByRole('heading', {
          name: title,
          level: 3
        })
        expect(heading).toBeInTheDocument()
      })
    })
  })

  describe('Navigation', () => {
    it('has correct navigation links', () => {
      const mainNav = screen.getByTestId('main-nav')
      expect(mainNav).toBeInTheDocument()
      expect(mainNav).toHaveAttribute('aria-label', '主导航')

      const headerLinks = [
        { text: '首页', href: '/' },
        { text: 'BMI计算器', href: '/health/bmi' },
        { text: '体脂计算器', href: '/calculators/body-fat' }
      ]
      
      headerLinks.forEach(({ text, href }) => {
        const link = within(mainNav).getByRole('link', { name: text })
        expect(link).toBeInTheDocument()
        expect(link).toHaveAttribute('href', href)
      })
    })
  })

  describe('Footer', () => {
    it('has correct footer navigation', () => {
      const footer = screen.getByTestId('main-footer')
      expect(footer).toBeInTheDocument()

      const quickLinksNav = within(footer).getByRole('navigation', { name: /快速链接/i })
      expect(quickLinksNav).toBeInTheDocument()

      const helpNav = within(footer).getByRole('navigation', { name: /帮助支持/i })
      expect(helpNav).toBeInTheDocument()
    })

    it('has correct footer links', () => {
      const footer = screen.getByTestId('main-footer')
      const quickLinksNav = within(footer).getByRole('navigation', { name: /快速链接/i })
      const quickLinks = [
        { text: '体脂计算器', href: '/calculators/body-fat' },
        { text: 'BMI计算器', href: '/calculators/bmi' },
        { text: '卡路里计算器', href: '/calculators/calorie' }
      ]
      
      quickLinks.forEach(({ text, href }) => {
        const link = within(quickLinksNav).getByRole('link', { name: text })
        expect(link).toBeInTheDocument()
        expect(link).toHaveAttribute('href', href)
      })

      const helpNav = within(footer).getByRole('navigation', { name: /帮助支持/i })
      const helpLinks = [
        { text: '常见问题', href: '/faq' },
        { text: '隐私政策', href: '/privacy' },
        { text: '使用条款', href: '/terms' }
      ]
      
      helpLinks.forEach(({ text, href }) => {
        const link = within(helpNav).getByRole('link', { name: text })
        expect(link).toBeInTheDocument()
        expect(link).toHaveAttribute('href', href)
      })
    })
  })

  describe('Accessibility', () => {
    it('has correct accessibility attributes', () => {
      const main = screen.getByRole('main')
      expect(main).toBeInTheDocument()

      const mainNav = screen.getByTestId('main-nav')
      expect(mainNav).toBeInTheDocument()
      expect(mainNav).toHaveAttribute('aria-label', '主导航')

      const footer = screen.getByTestId('main-footer')
      expect(footer).toBeInTheDocument()
      expect(footer).toHaveAttribute('role', 'contentinfo')

      const quickLinksNav = within(footer).getByRole('navigation', { name: /快速链接/i })
      expect(quickLinksNav).toBeInTheDocument()

      const helpNav = within(footer).getByRole('navigation', { name: /帮助支持/i })
      expect(helpNav).toBeInTheDocument()
    })
  })
}) 