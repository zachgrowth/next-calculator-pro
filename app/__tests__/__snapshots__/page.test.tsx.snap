// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page renders homepage unchanged 1`] = `
<div>
  <div
    class="min-h-screen bg-background"
  >
    <header
      class="sticky top-0 z-50 w-full border-b bg-background/95 backdrop-blur supports-[backdrop-filter]:bg-background/60"
    >
      <div
        class="container flex h-14 items-center"
      >
        <nav
          class="flex items-center space-x-6 text-sm font-medium"
        >
          <a
            class="transition-colors hover:text-foreground/80 text-foreground/60"
            href="/"
          >
            首页
          </a>
          <a
            class="transition-colors hover:text-foreground/80 text-foreground/60"
            href="/health/bmi"
          >
            BMI计算器
          </a>
          <a
            class="transition-colors hover:text-foreground/80 text-foreground/60"
            href="/calculators/body-fat"
          >
            体脂计算器
          </a>
        </nav>
      </div>
    </header>
    <main
      class="container mx-auto px-4"
    >
      <section
        class="py-16 md:py-24"
      >
        <div
          class="max-w-4xl mx-auto text-center space-y-8"
        >
          <h1
            class="text-4xl font-bold text-primary sm:text-5xl md:text-6xl bg-clip-text text-transparent bg-gradient-to-r from-primary to-primary/70 animate-gradient"
          >
            专业的健康计算工具
          </h1>
          <p
            class="text-xl text-muted-foreground max-w-2xl mx-auto"
          >
            帮助您更好地了解和管理身体健康，提供精确的数据分析和专业的健康建议
          </p>
          <div
            class="grid grid-cols-1 md:grid-cols-3 gap-6 mt-12 text-center"
          >
            <div
              class="p-6 rounded-lg bg-background border border-border"
            >
              <div
                class="w-12 h-12 rounded-full bg-primary/10 text-primary mx-auto flex items-center justify-center mb-4"
              >
                <svg
                  class="lucide lucide-calculator w-6 h-6"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="20"
                    rx="2"
                    width="16"
                    x="4"
                    y="2"
                  />
                  <line
                    x1="8"
                    x2="16"
                    y1="6"
                    y2="6"
                  />
                  <line
                    x1="16"
                    x2="16"
                    y1="14"
                    y2="18"
                  />
                  <path
                    d="M16 10h.01"
                  />
                  <path
                    d="M12 10h.01"
                  />
                  <path
                    d="M8 10h.01"
                  />
                  <path
                    d="M12 14h.01"
                  />
                  <path
                    d="M8 14h.01"
                  />
                  <path
                    d="M12 18h.01"
                  />
                  <path
                    d="M8 18h.01"
                  />
                </svg>
              </div>
              <h3
                class="text-lg font-semibold mb-2"
              >
                精确计算
              </h3>
              <p
                class="text-sm text-muted-foreground"
              >
                采用专业算法，确保计算结果的准确性
              </p>
            </div>
            <div
              class="p-6 rounded-lg bg-background border border-border"
            >
              <div
                class="w-12 h-12 rounded-full bg-primary/10 text-primary mx-auto flex items-center justify-center mb-4"
              >
                <svg
                  class="lucide lucide-heart w-6 h-6"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
                  />
                </svg>
              </div>
              <h3
                class="text-lg font-semibold mb-2"
              >
                健康建议
              </h3>
              <p
                class="text-sm text-muted-foreground"
              >
                提供个性化的健康建议和改善方案
              </p>
            </div>
            <div
              class="p-6 rounded-lg bg-background border border-border"
            >
              <div
                class="w-12 h-12 rounded-full bg-primary/10 text-primary mx-auto flex items-center justify-center mb-4"
              >
                <svg
                  class="lucide lucide-brain w-6 h-6"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.5 2A2.5 2.5 0 0 1 12 4.5v15a2.5 2.5 0 0 1-4.96.44 2.5 2.5 0 0 1-2.96-3.08 3 3 0 0 1-.34-5.58 2.5 2.5 0 0 1 1.32-4.24 2.5 2.5 0 0 1 1.98-3A2.5 2.5 0 0 1 9.5 2Z"
                  />
                  <path
                    d="M14.5 2A2.5 2.5 0 0 0 12 4.5v15a2.5 2.5 0 0 0 4.96.44 2.5 2.5 0 0 0 2.96-3.08 3 3 0 0 0 .34-5.58 2.5 2.5 0 0 0-1.32-4.24 2.5 2.5 0 0 0-1.98-3A2.5 2.5 0 0 0 14.5 2Z"
                  />
                </svg>
              </div>
              <h3
                class="text-lg font-semibold mb-2"
              >
                科学依据
              </h3>
              <p
                class="text-sm text-muted-foreground"
              >
                基于科学研究和专业标准制定
              </p>
            </div>
          </div>
        </div>
      </section>
      <section
        class="py-16 bg-gradient-to-b from-background to-background/50"
      >
        <div
          class="max-w-6xl mx-auto"
        >
          <h2
            class="text-3xl font-bold text-center mb-12"
          >
            选择计算工具
          </h2>
          <div
            class="grid gap-6 sm:grid-cols-2 lg:grid-cols-3"
          >
            <a
              class="group relative p-6 rounded-xl border border-border bg-card hover:shadow-lg transition-all duration-300 ease-in-out hover:scale-[1.02]"
              href="/calculators/body-fat"
            >
              <div
                class="w-12 h-12 rounded-full bg-rose-50 text-rose-500 flex items-center justify-center mb-4"
              >
                <svg
                  class="lucide lucide-heart w-6 h-6"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
                  />
                </svg>
              </div>
              <h3
                class="text-xl font-semibold mb-2 group-hover:text-primary transition-colors"
              >
                体脂计算器
              </h3>
              <p
                class="text-muted-foreground mb-4"
              >
                精确计算体脂率，了解身体成分
              </p>
              <div
                class="flex items-center text-sm text-primary"
              >
                <span>
                  开始使用
                </span>
                <svg
                  class="lucide lucide-arrow-right w-4 h-4 ml-2 group-hover:translate-x-1 transition-transform"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                  <path
                    d="m12 5 7 7-7 7"
                  />
                </svg>
              </div>
            </a>
            <a
              class="group relative p-6 rounded-xl border border-border bg-card hover:shadow-lg transition-all duration-300 ease-in-out hover:scale-[1.02]"
              href="/calculators/bmi"
            >
              <div
                class="w-12 h-12 rounded-full bg-blue-50 text-blue-500 flex items-center justify-center mb-4"
              >
                <svg
                  class="lucide lucide-calculator w-6 h-6"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="20"
                    rx="2"
                    width="16"
                    x="4"
                    y="2"
                  />
                  <line
                    x1="8"
                    x2="16"
                    y1="6"
                    y2="6"
                  />
                  <line
                    x1="16"
                    x2="16"
                    y1="14"
                    y2="18"
                  />
                  <path
                    d="M16 10h.01"
                  />
                  <path
                    d="M12 10h.01"
                  />
                  <path
                    d="M8 10h.01"
                  />
                  <path
                    d="M12 14h.01"
                  />
                  <path
                    d="M8 14h.01"
                  />
                  <path
                    d="M12 18h.01"
                  />
                  <path
                    d="M8 18h.01"
                  />
                </svg>
              </div>
              <h3
                class="text-xl font-semibold mb-2 group-hover:text-primary transition-colors"
              >
                BMI计算器
              </h3>
              <p
                class="text-muted-foreground mb-4"
              >
                评估身体质量指数，监控健康状态
              </p>
              <div
                class="flex items-center text-sm text-primary"
              >
                <span>
                  开始使用
                </span>
                <svg
                  class="lucide lucide-arrow-right w-4 h-4 ml-2 group-hover:translate-x-1 transition-transform"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                  <path
                    d="m12 5 7 7-7 7"
                  />
                </svg>
              </div>
            </a>
            <a
              class="group relative p-6 rounded-xl border border-border bg-card hover:shadow-lg transition-all duration-300 ease-in-out hover:scale-[1.02]"
              href="/calculators/calorie"
            >
              <div
                class="w-12 h-12 rounded-full bg-green-50 text-green-500 flex items-center justify-center mb-4"
              >
                <svg
                  class="lucide lucide-brain w-6 h-6"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.5 2A2.5 2.5 0 0 1 12 4.5v15a2.5 2.5 0 0 1-4.96.44 2.5 2.5 0 0 1-2.96-3.08 3 3 0 0 1-.34-5.58 2.5 2.5 0 0 1 1.32-4.24 2.5 2.5 0 0 1 1.98-3A2.5 2.5 0 0 1 9.5 2Z"
                  />
                  <path
                    d="M14.5 2A2.5 2.5 0 0 0 12 4.5v15a2.5 2.5 0 0 0 4.96.44 2.5 2.5 0 0 0 2.96-3.08 3 3 0 0 0 .34-5.58 2.5 2.5 0 0 0-1.32-4.24 2.5 2.5 0 0 0-1.98-3A2.5 2.5 0 0 0 14.5 2Z"
                  />
                </svg>
              </div>
              <h3
                class="text-xl font-semibold mb-2 group-hover:text-primary transition-colors"
              >
                卡路里计算器
              </h3>
              <p
                class="text-muted-foreground mb-4"
              >
                计算每日所需热量，科学饮食计划
              </p>
              <div
                class="flex items-center text-sm text-primary"
              >
                <span>
                  开始使用
                </span>
                <svg
                  class="lucide lucide-arrow-right w-4 h-4 ml-2 group-hover:translate-x-1 transition-transform"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                  <path
                    d="m12 5 7 7-7 7"
                  />
                </svg>
              </div>
            </a>
          </div>
        </div>
      </section>
    </main>
    <footer
      class="bg-muted mt-auto"
      role="contentinfo"
    >
      <div
        class="container mx-auto px-4 py-8"
      >
        <div
          class="grid grid-cols-1 md:grid-cols-4 gap-8"
        >
          <div
            class="space-y-4"
          >
            <h2
              class="text-lg font-semibold"
            >
              关于我们
            </h2>
            <p
              class="text-sm text-muted-foreground"
            >
              我们致力于提供准确、专业的健康计算工具，帮助您更好地了解和管理身体健康。
            </p>
          </div>
          <nav
            aria-label="快速链接"
            class="space-y-4"
          >
            <h2
              class="text-lg font-semibold"
            >
              快速链接
            </h2>
            <ul
              class="space-y-2"
            >
              <li>
                <a
                  class="text-sm text-muted-foreground hover:text-primary transition-colors"
                  href="/calculators/body-fat"
                >
                  体脂计算器
                </a>
              </li>
              <li>
                <a
                  class="text-sm text-muted-foreground hover:text-primary transition-colors"
                  href="/calculators/bmi"
                >
                  BMI计算器
                </a>
              </li>
              <li>
                <a
                  class="text-sm text-muted-foreground hover:text-primary transition-colors"
                  href="/calculators/calorie"
                >
                  卡路里计算器
                </a>
              </li>
            </ul>
          </nav>
          <nav
            aria-label="帮助支持"
            class="space-y-4"
          >
            <h2
              class="text-lg font-semibold"
            >
              帮助支持
            </h2>
            <ul
              class="space-y-2"
            >
              <li>
                <a
                  class="text-sm text-muted-foreground hover:text-primary transition-colors"
                  href="/faq"
                >
                  常见问题
                </a>
              </li>
              <li>
                <a
                  class="text-sm text-muted-foreground hover:text-primary transition-colors"
                  href="/privacy"
                >
                  隐私政策
                </a>
              </li>
              <li>
                <a
                  class="text-sm text-muted-foreground hover:text-primary transition-colors"
                  href="/terms"
                >
                  使用条款
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="space-y-4"
          >
            <h2
              class="text-lg font-semibold"
            >
              联系我们
            </h2>
            <address
              class="not-italic"
            >
              <ul
                class="space-y-2"
              >
                <li
                  class="text-sm text-muted-foreground"
                >
                  邮箱：
                  <a
                    class="hover:text-primary transition-colors"
                    href="mailto:contact@example.com"
                  >
                    contact@example.com
                  </a>
                </li>
                <li
                  class="text-sm text-muted-foreground"
                >
                  电话：
                  <a
                    class="hover:text-primary transition-colors"
                    href="tel:+8612345678900"
                  >
                    +86 123 4567 8900
                  </a>
                </li>
                <li
                  class="text-sm text-muted-foreground"
                >
                  地址：北京市朝阳区xxx街xxx号
                </li>
              </ul>
            </address>
          </div>
        </div>
        <div
          class="mt-8 pt-8 border-t text-center text-sm text-muted-foreground"
        >
          <p>
            © 
            2024
             健康计算器. 保留所有权利.
          </p>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
